bootloader/src/pyi_utils.c

```diff
@@ -740,6 +740,19 @@ pyi_utils_set_environment(const ARCHIVE_STATUS *status)
     return rc;
 }
 
+int
+set_systemd_env(pid_t pid)
+{
+    #define ULONG_STRING_SIZE (sizeof (unsigned long) * CHAR_BIT / 3 + 2)
+    const char * env_var = "LISTEN_PID";
+    char pid_str[ULONG_STRING_SIZE];
+    snprintf(pid_str, ULONG_STRING_SIZE, "%ld", (unsigned long)pid);
```

Comment:
Move this into the `if`. formatting is only required if the variable need to be set.